### ISSUE TITLE
colab error

### ISSUE THREAD

**Issue by yzhang-github-pub** [Shivanandroy/simpleT5#7]
When running google colab, the line below produced error:

model.load_model("t5","outputs/SimpleT5-epoch-2-train-loss-0.9526", use_gpu=True)

Error: 404 Client Error: Not Found for url: https://huggingface.co/outputs/SimpleT5-epoch-2-train-loss-0.9526/resolve/main/config.json

Please help. Thanks a lot.

**Comment by Shivanandroy** [comment]
Can you share the colab link?

**Comment by yzhang-github-pub** [comment]
Sure. Here it is:
https://colab.research.google.com/drive/1JZ8v9L0w0Ai3WbibTeuvYlytn0uHMP6O?usp=sharing#scrollTo=7PVmAPLUSBMh

I opened colab from your post at https://pythonrepo.com/repo/Shivanandroy-simpleT5.

**Comment by Shivanandroy** [comment]
The error you get is because the model path wasn't correct.
`model.load_model("t5","outputs/SimpleT5-epoch-2-train-loss-0.9526", use_gpu=True)`

Make sure, you have `outputs/SimpleT5-epoch-2-train-loss-0.9526` directory in colab.

**Comment by yzhang-github-pub** [comment]
Thank you so much @Shivanandroy! I really like your elegant solution to simplify fine tuning t5 models. This is particularly useful to end users without  a lot of experience in this area. Please keep the great work.

I also appreciate that you took time to check the error so quickly. To fix the error, I added one line in the notebook  before model.load_model(...) to capture the checkpoint from last epoch:

model_dir = !ls -rt1 outputs/SimpleT5* | grep outputs | tail -1 | cut -f1 -d':'

and modifications are made as in :

model.load_model("t5",model_dir[0], use_gpu=True)
 and  in
model.convert_and_load_onnx_model(model_dir=model_dir[0])

Now with raw model, model.predict(text_to_summarize) produced expected result.

However, I got an error in onnx conversion & loading:

Fail: [ONNXRuntimeError] : 1 : FAIL : Load model from /content/models/SimpleT5-epoch-2-train-loss-0-decoder-quantized.onnx failed:This is an invalid model. Error: Duplicate definition of name (pkv_11).

**Comment by Shivanandroy** [comment]
Thank you for your comment. Appreciated.

This is a known bug.
The latest version of simpleT5 uses `transformers == 4.8.2` to support ByT5 model. but this version doesn't support onnx inference. So, `model.convert_and_load_onnx_model(model_dir=model_dir[0])` will give you an error.

However you wish to run with onnx inference, you can use earlier version of SimpleT5 (`pip install simplet5==0.0.9`).
`model.convert_and_load_onnx_model(model_dir=model_dir[0])` will work just as expected.

Below is a complete example

```python
# pip install simplet5==0.0.9

from simplet5 import SimpleT5

# instantiate
model = SimpleT5()

# load (supports t5, mt5, byt5)
model.from_pretrained("t5","t5-base")

# train
model.train(train_df=train_df, # pandas dataframe with 2 columns: source_text & target_text
            eval_df=eval_df, # pandas dataframe with 2 columns: source_text & target_text
            source_max_token_len = 512, 
            target_max_token_len = 128,
            batch_size = 8,
            max_epochs = 5,
            use_gpu = True,
            outputdir = "outputs",
            early_stopping_patience_epochs = 0
            )

# load trained T5 model
model.load_model("t5","path/to/trained/model/directory", use_gpu=False)

# predict
model.predict("input text for prediction")

# need faster inference on CPU, get ONNX support
model.convert_and_load_onnx_model("path/to/T5 model/directory")
model.onnx_predict("input text for prediction")

**Comment by yucongo** [comment]
Since the loss may be slightly different in different run, that line in the colab notebook
`model.load_model("t5","outputs/SimpleT5-epoch-2-train-loss-0.9526", use_gpu=True)`
may be changed to
```
from pathlib import Path
model.load_model("t5", next(Path("outputs").glob("*epoch-2*")), use_gpu=True)
```
to automatically load the right directory. `next(Path("outputs").glob("*epoch-2*")` simply outputs the first item (if any) in "outputs/*epoch-2*".

Some checking can be made, for example
```
model_epoch2 = next(Path("outputs").glob("*epoch-2*")
if not model_epoch2.exists():
    print(f"{model_epoch2} does not exist.")
    raise SystemExit(1)
model.load_model("t5", model_epoch2, use_gpu=True)
```

Great repo by the way, thanks a lot.
